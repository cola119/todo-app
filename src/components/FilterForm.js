import React from 'react';

import FormControlLabel from '@material-ui/core/FormControlLabel';
import Switch from '@material-ui/core/Switch';

const FilterForm = (props) => {
    return (
        <>
            <FormControlLabel
                control={
                    <Switch
                        checked={props.filter['incomplete']}
                        onChange={() => props.handleFileter('incomplete')}
                        value="incomplete"
                    />
                }
                label="未完了のタスクを表示"
            />
            <FormControlLabel
                control={
                    <Switch
                        checked={props.filter['complete']}
                        onChange={() => props.handleFileter('complete')}
                        value="complete"
                    />
                }
                label="完了したタスクを表示"
            />
        </>
    );
};

export default FilterForm;