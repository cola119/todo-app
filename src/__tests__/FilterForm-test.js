import React from 'react';
import renderer from 'react-test-renderer';
import FilterForm from '../components/FilterForm';

test('<FilterForm />のスナップショット', () => {
    const stateLabels = {
        incomplete: "未完了",
        complete: "完了済み",
    }
    const todos = [
        { id: 12345, title: "title1", priority: "high", deadline: "today", state: 'incomplete', created_at: 1519211810362 },
        { id: 23456, title: "title2", priority: "medium", deadline: "tomorrow", state: 'complete', created_at: 1519211811670 },
        { id: 65432, title: "title3", priority: "low", deadline: "week", state: 'incomplete', created_at: 1519211809934 },
        { id: 83553, title: "title4", priority: "medium", deadline: "other", state: 'complete', created_at: 1519129853500 },
        { id: 53215, title: "title5", priority: "high", deadline: "week", state: 'incomplete', created_at: 1519129864400 },
    ];
    const stateFilter = { incomplete: true, complete: true };

    const tree = renderer.create(<FilterForm todos={todos} labels={stateLabels} labelKey="state" filter={stateFilter} />).toJSON();
    expect(tree).toMatchSnapshot();
})