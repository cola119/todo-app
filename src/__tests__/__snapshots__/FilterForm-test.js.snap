// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterForm />のスナップショット 1`] = `
Array [
  <label
    className="MuiFormControlLabel-root-1"
  >
    <span
      className="MuiButtonBase-root-23 MuiIconButton-root-17 MuiPrivateSwitchBase-root-13 MuiCheckbox-root-7 MuiCheckbox-colorPrimary-11 MuiPrivateSwitchBase-checked-14 MuiCheckbox-checked-8"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label-22"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-26"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <input
          checked={true}
          className="MuiPrivateSwitchBase-input-16"
          data-indeterminate={false}
          onChange={[Function]}
          type="checkbox"
          value="未完了"
        />
      </span>
      <span
        className="MuiTouchRipple-root-71"
      />
    </span>
    <span
      className="MuiTypography-root-35 MuiTypography-body1-44 MuiFormControlLabel-label-6"
    >
      未完了(3)
    </span>
  </label>,
  <label
    className="MuiFormControlLabel-root-1"
  >
    <span
      className="MuiButtonBase-root-23 MuiIconButton-root-17 MuiPrivateSwitchBase-root-13 MuiCheckbox-root-7 MuiCheckbox-colorPrimary-11 MuiPrivateSwitchBase-checked-14 MuiCheckbox-checked-8"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label-22"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-26"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
          />
        </svg>
        <input
          checked={true}
          className="MuiPrivateSwitchBase-input-16"
          data-indeterminate={false}
          onChange={[Function]}
          type="checkbox"
          value="完了済み"
        />
      </span>
      <span
        className="MuiTouchRipple-root-71"
      />
    </span>
    <span
      className="MuiTypography-root-35 MuiTypography-body1-44 MuiFormControlLabel-label-6"
    >
      完了済み(2)
    </span>
  </label>,
]
`;
